js/base/Exchange.js

```diff
@@ -1867,7 +1867,7 @@ module.exports = class Exchange {
         for (let i = 0; i < response.length; i++) {
             const item = response[i];
             const id = this.safeString (item, marketIdKey);
-            const market = this.safeMarket (id);
+            const market = this.safeMarket (id, undefined, undefined, this.safeString (this.options, 'defaultType'));
```

Comment:
@frosty00 that's why I think we should default to `defaultType` inside `safeMarket`, we end up having to do this, which is virtually the same